File: nodejs/sample_script.js
/**
 * sample_script.js
 *
 * A script demonstrating how to use the bandwidthbxml package
 *
 * NOTE: This script WILL NOT run in the project repo. You will need to copy it into your
 * own project and install "bandwidthbxml"
 *
 * @author Jacob Mulford
 */

var BxmlBuilder = require("bandwidthbxml");

//Hangup response
var hangup = new BxmlBuilder.Verbs.Hangup();

var response = new BxmlBuilder.Response();
response.addVerb(hangup);

console.log(response.toXml());

//PlayAudio response
var playAudio = new BxmlBuilder.Verbs.PlayAudio();
playAudio.setUrl("https://test.com");
playAudio.setUsername("user");
playAudio.setPassword("pass");

var response = new BxmlBuilder.Response();
response.addVerb(playAudio);

console.log(response.toXml());

//SpeakSentence response
var speakSentence = new BxmlBuilder.Verbs.SpeakSentence();
speakSentence.setSentence("test");
speakSentence.setVoice("susan");
speakSentence.setGender("female");
speakSentence.setLocale("en_US");

var response = new BxmlBuilder.Response();
response.addVerb(speakSentence);

console.log(response.toXml());

//SendDtmf response
var sendDtmf = new BxmlBuilder.Verbs.SendDtmf();
sendDtmf.setDtmf("123");

var response = new BxmlBuilder.Response();
response.addVerb(sendDtmf);

console.log(response.toXml());

//Forward response
var forward = new BxmlBuilder.Verbs.Forward();
forward.setTo("+18888888888");
forward.setFrom("+19999999999");
forward.setCallTimeout(3);
forward.setDiversionTreatment("none");
forward.setDiversionReason("away");

var response = new BxmlBuilder.Response();
response.addVerb(forward);

console.log(response.toXml());

//Pause response
var pause = new BxmlBuilder.Verbs.Pause();
pause.setDuration(3);

var response = new BxmlBuilder.Response();
response.addVerb(pause);

console.log(response.toXml());

//Redirect response
var redirect = new BxmlBuilder.Verbs.Redirect();
redirect.setUsername("user");
redirect.setPassword("pass");
redirect.setRedirectUrl("https://test.com");
redirect.setRedirectMethod("GET");
redirect.setTag("tag");

var response = new BxmlBuilder.Response();
response.addVerb(redirect);

console.log(response.toXml());

//Gather response with a nested SpeakSentence
var speakSentence = new BxmlBuilder.Verbs.SpeakSentence();
speakSentence.setSentence("test");
speakSentence.setVoice("susan");
speakSentence.setGender("female");
speakSentence.setLocale("en_US");

var gather = new BxmlBuilder.Verbs.Gather();
gather.setGatherUrl("https://test.com");
gather.setGatherMethod("GET");
gather.setUsername("user");
gather.setPassword("pass");
gather.setTag("tag");
gather.setTerminatingDigits("123");
gather.setMaxDigits(3);
gather.setInterDigitTimeout(4);
gather.setFirstDigitTimeout(5);
gather.setSpeakSentence(speakSentence);

var response = new BxmlBuilder.Response();
response.addVerb(gather);

console.log(response.toXml());

//Gather response with a nested PlayAudio
var playAudio = new BxmlBuilder.Verbs.PlayAudio();
playAudio.setUrl("https://test.com");
playAudio.setUsername("user");
playAudio.setPassword("pass");

var gather = new BxmlBuilder.Verbs.Gather();
gather.setGatherUrl("https://test.com");
gather.setGatherMethod("GET");
gather.setUsername("user");
gather.setPassword("pass");
gather.setTag("tag");
gather.setTerminatingDigits("123");
gather.setMaxDigits(3);
gather.setInterDigitTimeout(4);
gather.setFirstDigitTimeout(5);
gather.setPlayAudio(playAudio);

var response = new BxmlBuilder.Response();
response.addVerb(gather);

console.log(response.toXml());

//Gather response with no nested verbs
var gather = new BxmlBuilder.Verbs.Gather();
gather.setGatherUrl("https://test.com");
gather.setGatherMethod("GET");
gather.setUsername("user");
gather.setPassword("pass");
gather.setTag("tag");
gather.setTerminatingDigits("123");
gather.setMaxDigits(3);
gather.setInterDigitTimeout(4);
gather.setFirstDigitTimeout(5);
gather.setRepeatCount(6);

var response = new BxmlBuilder.Response();
response.addVerb(gather);

console.log(response.toXml());

//Transfer response
var number1 = new BxmlBuilder.Verbs.PhoneNumber();
number1.setNumber("+17777777777");
number1.setTransferAnswerUrl("https://test.com");
number1.setTransferAnswerMethod("GET");
number1.setUsername("user");
number1.setPassword("pass");
number1.setTag("tag");

var number2 = new BxmlBuilder.Verbs.PhoneNumber();
number2.setNumber("+17777777779");
number2.setTransferAnswerUrl("https://test2.com");
number2.setTransferAnswerMethod("POST");
number2.setUsername("user2");
number2.setPassword("pass2");
number2.setTag("tag2");

var transfer = new BxmlBuilder.Verbs.Transfer();
transfer.setTransferCallerId("+18888888888");
transfer.setCallTimeout(3);
transfer.setTag("tagTransfer");
transfer.setTransferCompleteUrl("https://testtransfer.com");
transfer.setTransferCompleteMethod("GET");
transfer.setUsername("usertransfer");
transfer.setPassword("passtransfer");
transfer.setDiversionTreatment("none");
transfer.setDiversionReason("away");
transfer.addPhoneNumber(number1);
transfer.addPhoneNumber(number2);

var response = new BxmlBuilder.Response();
response.addVerb(transfer);

console.log(response.toXml());

//PlayAudio and SpeakSentence together response
var playAudio = new BxmlBuilder.Verbs.PlayAudio();
playAudio.setUrl("https://test.com");
playAudio.setUsername("user");
playAudio.setPassword("pass");

var speakSentence = new BxmlBuilder.Verbs.SpeakSentence();
speakSentence.setSentence("test");
speakSentence.setVoice("susan");
speakSentence.setGender("female");
speakSentence.setLocale("en_US");

var response = new BxmlBuilder.Response();
response.addVerb(playAudio);
response.addVerb(speakSentence);

console.log(response.toXml());

File: ruby/sample_script.rb
=begin
sample_script.rb

A script that shows how to use the RubyBandwidthBxml package to create BXML

@author Jacob Mulford
@copyright Bandwidth INC
=end
require_relative "lib/ruby_bandwidth_bxml"

### Create a hangup response ###
response = RubyBandwidthBxml::Response.new()
hangup = RubyBandwidthBxml::Hangup.new()

response.push(hangup)
puts response.to_xml()

### Create a forward response ###
response = RubyBandwidthBxml::Response.new()
forward = RubyBandwidthBxml::Forward.new({
    :to => "+19999999999",
    :from => "+19999999999",
    :call_timeout => "3",
    :diversion_treatment => "none",
    :diversion_reason => "away"
})

response.push(forward)
puts response.to_xml()

### Create a pause response ###
response = RubyBandwidthBxml::Response.new()
pause = RubyBandwidthBxml::Pause.new({
    :duration => 3
})

response.push(pause)
puts response.to_xml()

### Create a redirect response ###
response = RubyBandwidthBxml::Response.new()
redirect = RubyBandwidthBxml::Redirect.new({
    :redirect_url => "https://test.com",
    :redirect_method => "GET",
    :tag => "tag",
    :username => "user",
    :password => "pass"
})

response.push(redirect)
puts response.to_xml()

### Create a transfer response ###
response = RubyBandwidthBxml::Response.new()
phone_number = RubyBandwidthBxml::PhoneNumber.new({
    :number => "+19999999999",
    :transfer_answer_url => "https://test.com",
    :transfer_answer_method => "GET",
    :username => "user",
    :password => "pass"
})
transfer = RubyBandwidthBxml::Transfer.new({
    :transfer_caller_id => "+19999999999",
    :call_timeout => "3",
    :tag => "tag",
    :transfer_complete_url => "https://test.com",
    :transfer_complete_method => "GET",
    :username => "user",
    :password => "pass",
    :diversion_treatment => "none",
    :diversion_reason => "away",
    :phone_numbers => [phone_number]
})

response.push(transfer)
puts response.to_xml()

### Create a speak sentence response ###
response = RubyBandwidthBxml::Response.new()
speak_sentence = RubyBandwidthBxml::SpeakSentence.new({
    :sentence => "Test",
    :voice => "susan",
    :locale => "en_US",
    :gender => "female"
})

response.push(speak_sentence)
puts response.to_xml()

### Create a play audio response ###
response = RubyBandwidthBxml::Response.new()
play_audio = RubyBandwidthBxml::PlayAudio.new({
    :url => "https://test.mp3",
    :username => "user",
    :password => "pass"
})

response.push(play_audio)
puts response.to_xml()

### Create a gather response ###
response = RubyBandwidthBxml::Response.new()
gather = RubyBandwidthBxml::Gather.new({
    :gather_url => "https://test.com",
    :gather_method => "GET",
    :terminating_digits => "1",
    :tag => "tag",
    :max_digits => "3",
    :inter_digit_timeout => "3",
    :username => "username",
    :password => "password",
    :first_digit_timeout => "3",
    :repeat_count => "4"
})

#Gathers can be optionally nested with one of the PlayAudio or SpeakSentence verbs
speak_sentence = RubyBandwidthBxml::SpeakSentence.new({
    :sentence => "Test",
    :voice => "susan",
    :locale => "en_US"
})
gather_nested_speak_sentence = RubyBandwidthBxml::Gather.new({
    :gather_url => "https://test.com",
    :gather_method => "GET",
    :terminating_digits => "1",
    :tag => "tag",
    :max_digits => "3",
    :inter_digit_timeout => "3",
    :username => "username",
    :password => "password",
    :first_digit_timeout => "3",
    :speak_sentence => speak_sentence
})

play_audio = RubyBandwidthBxml::PlayAudio.new({
    :url => "https://test.mp3",
    :username => "user",
    :password => "pass"
})
gather_nested_play_audio = RubyBandwidthBxml::Gather.new({
    :gather_url => "https://test.com",
    :gather_method => "GET",
    :terminating_digits => "1",
    :tag => "tag",
    :max_digits => "3",
    :inter_digit_timeout => "3",
    :username => "username",
    :password => "password",
    :first_digit_timeout => "3",
    :play_audio => play_audio
})

response.push(gather)
response.push(gather_nested_speak_sentence)
response.push(gather_nested_play_audio)
puts response.to_xml()

### Create a send dtmf response ###
response = RubyBandwidthBxml::Response.new()
send_dtmf = RubyBandwidthBxml::SendDtmf.new({
    :dtmf => "123"
})

response.push(send_dtmf)
puts response.to_xml()

### Create a response that has speak sentence and play audio tags ###
response = RubyBandwidthBxml::Response.new()
speak_sentence = RubyBandwidthBxml::SpeakSentence.new({
    :sentence => "Test",
    :voice => "susan",
    :locale => "en_US",
    :gender => "female"
})

play_audio = RubyBandwidthBxml::PlayAudio.new({
    :url => "https://test.mp3",
    :username => "user",
    :password => "pass"
})

response.push(speak_sentence)
response.push(play_audio)
puts response.to_xml()
